core/src/main/scala/org/apache/spark/executor/ProcfsMetricsGetter.scala

```diff
@@ -64,19 +74,11 @@ private[spark] class ProcfsMetricsGetter(procfsDir: String = "/proc/") extends L
         SparkEnv.get.conf.get(config.EVENT_LOG_PROCESS_TREE_METRICS)
       val shouldAddProcessTreeMetricsToMetricsSet =
         SparkEnv.get.conf.get(config.METRICS_PROCESS_TREE_METRICS)
-      val pickEitherUIOrMetricsSet = shouldLogStageExecutorProcessTreeMetrics ^
-        shouldAddProcessTreeMetricsToMetricsSet
-      val areBothUIMetricsEnabled = shouldLogStageExecutorProcessTreeMetrics &&
+      val pickAnyOfUIOrMetricsSet = shouldLogStageExecutorProcessTreeMetrics ||
         shouldAddProcessTreeMetricsToMetricsSet
-      if (areBothUIMetricsEnabled) {
-        logWarning("You have enabled " +
-          "both spark.eventLog.logStageExecutorProcessTreeMetrics.enabled" +
-          " and spark.metrics.logStageExecutorProcessTreeMetrics.enabled. This isn't " +
-          "allowed. As a result Procfs metrics won't be reported to UI or Metricsset")
-      }
-      (procDirExists.get && shouldLogStageExecutorMetrics && pickEitherUIOrMetricsSet) ||
-        (procDirExists.get && !shouldLogStageExecutorMetrics &&
-          pickEitherUIOrMetricsSet && shouldAddProcessTreeMetricsToMetricsSet)
+
+      (procDirExists.get && shouldLogStageExecutorMetrics && pickAnyOfUIOrMetricsSet) ||
```

Comment:
I think the return statement can be further simplified to just `procDirExists.get && pickAnyOfUIOrMetricsSet`

Reply:
Not really. The procfs metrics shouldn't be reported to REST API or UI if the stage executor metrics aren't enables, but they can be reported to metrics system if it is enabled.

Reply:
Right, isn't that what `pickAnyOfUIOrMetricsSet` is for?

Reply:
pickAnyOfUIOrMetricsSet is just for checking whether one of UI or metrics set is enabled. Enabling stage level metrics is through the other metric If we want process tree metrics to be reported the stage level executor metrics which report metrics over heartbeat should also be enabled.

Reply:
I think I can remove the checks from here per your suggestion and just do a little update to one od the tests in ExecutorMetricsType. I will do more testing to avoid mistakes. There are 8 possible configs that 6 of them are acceptable but two are not.